common/include/pcl/common/tuple_utils.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *
+ */
+
+#pragma once
+
+#include <pcl/type_traits.h>
+
+#include <type_traits>
+#include <tuple>
+
+namespace pcl {
+
+// for_each_tuple_until
+// Iterate over tuple
+// https://stackoverflow.com/questions/26902633/how-to-iterate-over-a-stdtuple-in-c-11
+template <typename TupleType, typename FunctionType>
+void
+for_each_until_true(
+    TupleType&&,
+    FunctionType,
+    std::integral_constant<
+        std::size_t,
+        std::tuple_size<typename std::remove_reference<TupleType>::type>::value>)
+{}
+
+template <std::size_t I,
```

Comment:
The order feels a bit weird. Should not `I` come after `FunctionType`? Why make it the first item?

Reply:
Fixed. Had overlooked it.